tests/unit/test_credentials.py

```diff
@@ -3128,6 +3134,61 @@ def test_can_pass_basic_auth_token(self):
         self.assertEqual(creds.token, 'token')
         self.assertEqual(creds.method, 'container-role')
 
+    def test_can_pass_auth_token_from_file(self):
+        token_file_path = self.tempdir + '/token.jwt'
+        token_file = open(token_file_path, "x")
+        token_file.write("Basic auth-token")
+        token_file.close()
+        environ = {
+            'AWS_CONTAINER_CREDENTIALS_FULL_URI': 'http://localhost/foo',
+            'AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE': token_file_path,
+        }
+        fetcher = self.create_fetcher()
+        timeobj = datetime.now(tzlocal())
+        timestamp = (timeobj + timedelta(hours=24)).isoformat()
+        fetcher.retrieve_full_uri.return_value = {
+            "AccessKeyId": "access_key",
+            "SecretAccessKey": "secret_key",
+            "Token": "token",
+            "Expiration": timestamp,
+        }
+        provider = credentials.ContainerProvider(environ, fetcher)
+        creds = provider.load()
+
+        fetcher.retrieve_full_uri.assert_called_with(
+            'http://localhost/foo',
+            headers={'Authorization': 'Basic auth-token'},
+        )
+        self.assertEqual(creds.access_key, 'access_key')
+        self.assertEqual(creds.secret_key, 'secret_key')
+        self.assertEqual(creds.token, 'token')
+        self.assertEqual(creds.method, 'container-role')
+
+    def test_throws_error_on_invalid_token_file(self):
+        token_file_path = '/some/path/token.jwt'
+        environ = {
+            'AWS_CONTAINER_CREDENTIALS_FULL_URI': 'http://localhost/foo',
+            'AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE': token_file_path,
+        }
+        fetcher = self.create_fetcher()
+        provider = credentials.ContainerProvider(environ, fetcher)
+
+        with self.assertRaises(FileNotFoundError):
+            provider.load()
```

Comment:
Is this test necessary?

Reply:
Debatable; it's part of the specification that an error needs to be thrown here, but it should always be thrown by the open() method, and maybe this test should be added if we ever use something other than open, but for now this test is not necessary?  

I'll leave it in for now, but will likely delete it before this gets merged